source/utilities.cc

```diff
@@ -1093,12 +1234,16 @@ namespace aspect
 
               data_string = urlString.str();
               filesize = data_string.size();
-              delete url;
+
 #else // ASPECT_WITH_LIBDAP
 
-              // broadcast failure state, then throw
-              const int ierr = MPI_Bcast(&filesize, 1, MPI_UNSIGNED, 0, comm);
-              AssertThrowMPI(ierr);
+              // Broadcast failure state, then throw. We signal the failure by
+              // setting the file size to an invalid size, then trigger an assert.
+              {
+                std::size_t invalid_filesize = numbers::invalid_unsigned_int;
```

Comment:
shouldn't this be an invalid size_t? You can use -1 directly.

Reply:
Ah, I had not realized that we actually have `invalid_size_type`. So fixed.